src/components/calendar/nz-calendar.component.ts

```diff
@@ -334,7 +339,8 @@ export class NzCalendarComponent implements OnInit {
         index          : i,
         name           : this._listOfMonthName[ i ],
         isCurrentMonth : moment(new Date()).month() === i && date.isSame(new Date(), 'year'),
-        isSelectedMonth: this._showMonth === i
+        isSelectedMonth: this._showMonth === i,
+        disabled       : this.nzDisabledDate && this.nzDisabledDate(date.month(i).toDate(), 'month')
```

Comment:
why need to add param 'month' here? 
should nzDisabledDate function work same for both day and month mode?

Reply:
because can judgment in day mode. the month picker only needs to be accurate to month.
like [this](https://github.com/NG-ZORRO/ng-zorro-antd/pull/451/files#diff-c5165b47301db0f2c2bbc889ef0f17feR41)

Reply:
@HsuanXyz but the function works for day mode also works for month mode, is that right?

Reply:
yes

Reply:
I just think this design is somewhat redundant

Reply:
Because in the date range demo, if the end of the date part is greater than the start. will be wrongly disabled.
eg: `2017-10-20 20:00` - `2017-11-20 10:00` The end date of October will be disabled

Reply:
this is better?
```diff
- disabled : this.nzDisabledDate && this.nzDisabledDate(date.month(i).toDate(), 'month')
+ disabled : this.nzDisabledDate && this.nzDisabledDate(date.month(i).day(0).toDate())
```

Reply:
it should not disable the month when the disable function across the month range.

Reply:
agree, thank for you suggestion.